### PR TITLE
docs: Add AppMenu

### PR DESCRIPTION
**Pull request by wegry**
Related to InboxSDK/InboxSDK#932

### CODE REVIEW

File: docs/appmenu.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+---
+title: AppMenu
+---
+
+This namespace contains functionality associated with adding app menu items to the app menu panel of Gmail. Typically, these app menu items are accessible by the user on the left side of the email client and include built in app menu items like the Inbox and optionally either Chat, Meet, or both. If both Chat and Meet aren't enabled, the AppMenu will not be displayed.
```
Comment (aleemstreak) [inline docs/appmenu.md:5]:
```suggestion
This namespace contains functionality associated with adding app menu items to the app menu panel of Gmail. Typically, these app menu items are accessible by the user on the left side of the email client and include built in app menu items like the Mail and optionally either Chat, Meet, or both. If both Chat and Meet aren't enabled, the AppMenu will not be displayed.
```

File: docs/appmenu.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+This namespace allows you to add your own items to this App menu. Typically, these app menu items are useful to either place high level collapsible panels like the NavMenu or send users to different Routes that you have already registered providing navigation for your entire application.
+The app menu is represented as a list of items. Each item can have either a CollapsiblePanel or click handler.
+Items with CollapsiblePanels can also have accessories which provide primary actions like providing a "create new" action.
+
+### Namespace methods
+
+#### isShown()
+
+Returns true if the user has Chat, Meet, or both enabled.
```
Comment (aleemstreak) [inline docs/appmenu.md:15]:
```suggestion
Returns true if the app menu is visible. At time of writing, this typically means they have chose to enable Chat, Meet, or both.
```

File: docs/appmenu.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+#### addMenuItem(appMenuItemDescriptor)
+
+Adds an app menu item to the app menu. Because the AppMenu may not be shown, this method expects `AppMenu.isShown` to be called first. If the `insertOrder` option is provided, the app menu item will be added at that index. If it is not, the SDK adds the app menu item after Gmail's default app menu icons.
```
Comment (aleemstreak) [inline docs/appmenu.md:19]: what happens if you call this when the app menu isn't shown?

Comment (wegry) [reply docs/appmenu.md:19]: To the UI? Nothing. We look for the selector of the parent of the app menu, and if said selectors's not there after waiting for it, we time out.

The SDK will log an error in that case. Will add a note here.

**Pull request merged**